### PR TITLE
the 3.2.x release of exporting app bundles is very simple

### PR DESCRIPTION
**Pull request by amanj120**

### CODE REVIEW

File: platform/android/export/export.cpp [hunk 1]
```diff
@@ -2101,13 +2102,23 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			}
 		}
 
+		if (bool(p_preset->get("custom_template/export_app_bundle"))) {
```
Comment (thanwah) [inline platform/android/export/export.cpp]: `if (bool(p_preset->get("custom_template/export_app_bundle")) && !bool(p_preset->get("custom_template/use_custom_build"))) {`

File: platform/android/export/export.cpp [hunk 2]
```diff
@@ -2101,13 +2102,23 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
...
+
 		r_error = err;
 		return valid;
 	}
 
 	virtual List<String> get_binary_extensions(const Ref<EditorExportPreset> &p_preset) const {
 		List<String> list;
 		list.push_back("apk");
+		list.push_back("aab");
+		//TODO: make sure "aab" is selected only when "export_app_bundle" is selected.
```
Comment (thanwah) [inline platform/android/export/export.cpp]: Can you just add the check here? Or is this function called and its result cached across toggling of the AAB checkbox?

File: platform/android/export/export.cpp [hunk 3]
```diff
@@ -2587,6 +2620,26 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 				EditorNode::get_singleton()->show_warning(TTR("Building of Android project failed, check output for the error.\nAlternatively visit docs.godotengine.org for Android build documentation."));
 				return ERR_CANT_CREATE;
 			}
+
+			if (export_app_bundle) {
+				List<String> copy_args;
+				String output_path;
```
Comment (thanwah) [inline platform/android/export/export.cpp]: Suggest something similar to

String build_type = (p_debug ? "Debug" : "Release");
String build_command = vformat("bundle%s", build_type);

File: platform/android/export/export.cpp [hunk 4]
```diff
@@ -2566,7 +2591,15 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			if (clean_build_required) {
 				cmdline.push_back("clean");
 			}
-			cmdline.push_back("build");
+
+			if (export_app_bundle) {
+				String build_type = (p_debug ? "Debug" : "Release");
```
Comment (thanwah) [inline platform/android/export/export.cpp]: `String build_type = p_debug ? "Debug" : "Release";` is my personal preference

**Review by thanwah — CHANGES_REQUESTED** [review]

File: platform/android/export/export.cpp (hunk 2 above)
Comment (amanj120) [reply platform/android/export/export.cpp]: All this does is adds ".aab" as a valid choice for the filename extension. The check has to be made in the `export_project` function, but I'm still figuring out where to put it.

Comment (amanj120) [reply platform/android/export/export.cpp]: I think I took care of this now [here](https://github.com/amanj120/godot/pull/18/files#diff-fb88394f2df1faf844deaff3b3639aeeR2468)

**Pull request closed without merging**